ats/src/ats/attributedict.py

```diff
@@ -31,33 +28,21 @@ def __delattr__(self, name):
             raise AttributeError('No attribute %s' % name)
 
     def __repr__(self):
-        out = StringIO()
-        print("AttributeDict(", file=out)
-        keys = self.keys()
-        keys.sort()
-        for key in keys:
-            print("  %s = %r," % (key, self[key]), file=out)
-        print(")", file=out)
-        s = out.getvalue()
-        out.close()
-        return s
+        _repr = "AttributeDict(\n"
+        for key in sorted(self.keys()):
+            _repr += " %s = %r,\n" % (key, self[key])
+        _repr += ")"
+        return _repr
 
     def __str__(self):
         "Prints the attributes in alphabetical order."
```

Comment:
Aren't docstrings supposed to have three quotes? (out of scope of pr)

Reply:
Yes they should have three quotes. [PEP 257](https://www.python.org/dev/peps/pep-0257/)
I created [an issue](https://github.com/LLNL/ATS/issues/29) for this so it is not forgotten.